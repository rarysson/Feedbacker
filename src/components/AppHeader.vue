<template>
  <div class="flex justify-between items-center w-4/5 max-w-6xl py-10">
    <div class="w-28 lg:w-36">
      <img class="w-full" src="@/assets/image/logo.png" alt="logo" />
    </div>
    <div class="flex">
      <ul class="flex list-none">
        <li
          class="
            px-6
            py-2
            mr-2
            font-bold
            text-white
            rounded-full
            cursor-pointer
            focus:outline-none
          "
          @click="() => router.push({ name: 'Credentials' })"
        >
          Credenciais
        </li>
        <li
          class="
            px-6
            py-2
            mr-2
            font-bold
            text-white
            rounded-full
            cursor-pointer
            focus:outline-none
          "
          @click="() => router.push({ name: 'Feedbacks' })"
        >
          Feedbacks
        </li>
        <li
          class="
            px-6
            py-2
            font-bold
            bg-white
            rounded-full
            text-brand-main
            cursor-pointer
            focus:outline-none
          "
          data-test="logout-btn"
          @click="handleLogout"
        >
          {{ logoutLabel }}
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
import { useRouter } from "vue-router";
import { computed } from "@vue/reactivity";

import useStore from "@/hooks/useStore";
import { clearCurrentUser } from "../store/user";

export default {
  name: "AppHeader",

  setup() {
    const router = useRouter();
    const store = useStore("User");

    const logoutLabel = computed(() => {
      if (!store.currentUser.name) {
        return "...";
      }

      return `${store.currentUser.name} (sair)`;
    });

    function handleLogout() {
      window.localStorage.removeItem("token");
      clearCurrentUser();
      router.push({ name: "Home" });
    }

    return {
      router,
      logoutLabel,
      handleLogout,
    };
  },
};
</script>
