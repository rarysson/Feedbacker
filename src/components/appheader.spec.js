import { routes } from "../router";
import { shallowMount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";

import AppHeader from "./AppHeader.vue";

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

const mockStore = { currentUser: {} };

jest.mock("../hooks/useStore", () => {
  return () => {
    return mockStore;
  };
});

describe("AppHeader", () => {
  it("deveria renderizar header do app", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = shallowMount(AppHeader, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("deveria aparecer os 3 pontos quando o usuário não está logado", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = shallowMount(AppHeader, {
      global: {
        plugins: [router],
      },
    });

    const buttonLogout = wrapper.find('[data-test="logout-btn"]');
    expect(buttonLogout.text()).toBe("...");
  });

  it("deveria aparecer o nome do usuário quando o usuário está logado", async () => {
    router.push("/");
    await router.isReady();
    mockStore.currentUser.name = "Test";
    const wrapper = shallowMount(AppHeader, {
      global: {
        plugins: [router],
      },
    });

    const buttonLogout = wrapper.find('[data-test="logout-btn"]');
    expect(buttonLogout.text()).toBe("Test (sair)");
  });
});
