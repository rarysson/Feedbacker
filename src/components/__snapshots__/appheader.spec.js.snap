// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader deveria renderizar header do app 1`] = `
<div class="flex justify-between items-center w-4/5 max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img class="w-full" src="" alt="logo"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li class="
            px-6
            py-2
            mr-2
            font-bold
            text-white
            rounded-full
            cursor-pointer
            focus:outline-none
          "> Credenciais </li>
      <li class="
            px-6
            py-2
            mr-2
            font-bold
            text-white
            rounded-full
            cursor-pointer
            focus:outline-none
          "> Feedbacks </li>
      <li class="
            px-6
            py-2
            font-bold
            bg-white
            rounded-full
            text-brand-main
            cursor-pointer
            focus:outline-none
          " data-test="logout-btn">...</li>
    </ul>
  </div>
</div>
`;
